vnext/Shared/Threading/BatchingQueueThread.cpp

```diff
@@ -71,8 +71,15 @@ void BatchingQueueThread::decoratedNativeCallInvokerReady(
   std::scoped_lock lck(m_mutex);
   m_callInvoker->invokeAsync([wkInstance, this] {
     if (auto instance = wkInstance.lock()) {
-      std::scoped_lock lck(m_mutex);
-      m_callInvoker = instance->getDecoratedNativeCallInvoker(m_callInvoker);
+      std::scoped_lock lckQuitting(m_mutexQuitting);
+
+      // If we are shutting down, then then the mutex is being held in quitSynchronous
```

Comment:
> // If we are shutting down, [](http://example.com/codeflow?start=5&length=28)

Consider using the std::recursive_mutex to avoid issues with the std::mutex m_mutex in the nested call cases.

Reply:
Keeping it the same as the master change that I'm porting - to avoid additional risk to the backport.